connectors/utils.py

```diff
@@ -401,3 +401,26 @@ async def func_to_execute(*args, **kwargs):
 
 def e2str(entry):
     return entry.name.lower()
+
+
+class MultiService:
+    def __init__(self, services):
+        self._services = services
+
+    async def run(self):
+        tasks = [asyncio.create_task(service.run()) for service in self._services]
+
+        done, pending = await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
+
+        for task in pending:
+            task.cancel()
+            await task
```

Comment:
you want:

```
try:
    await task
except asyncio.CancelledError:
    logger.error('...')
```

to make sure you don't break in this loop if the task did not handle the cancellation